src/dfx/src/lib/builders/mod.rs

```diff
@@ -376,6 +376,23 @@ impl BuildConfig {
     }
 }
 
+fn shrink_wasm(wasm_path: impl AsRef<Path>) -> DfxResult {
+    let wasm_path = wasm_path.as_ref();
+    let wasm = std::fs::read(wasm_path).expect("Could not read the WASM module.");
+    let mut module_config = walrus::ModuleConfig::new();
+    module_config.generate_name_section(true);
+    module_config.generate_producers_section(false);
+    let mut module = module_config
+        .parse(&wasm)
+        .expect("Could not parse the WASM module.");
+    ic_wasm::shrink::shrink(&mut module);
+    module
+        .emit_wasm_file(wasm_path)
+        .unwrap_or_else(|_| panic!("Could not write shrinked WASM to {:?}", wasm_path));
```

Comment:
Same here, unless there's something I'm missing, we should propagate the error rather than panic
```suggestion
        .with_context(|| format!("Could not write shrinked WASM to {:?}", wasm_path))?;
```